Release 1.4 [TypeError: Cannot read property 'name' of undefined] when a peer disconnects from the network

rbntt opened the pull request:
I am testing high availability for Hyperledger Fabric.

I am manually enabling/disabling peer nodes when I insert data into the blockchain and I expect my Node SDK client to send me messages of impossible connection.

This correctly happens when your peer node is down and your Node SDK tries the connection. In this case, the SDK says 'Unable to initialize the channel'. But this not happens if you disable a node at runtime.

In that situation, I receive this error:

 

"Cannot read property 'name' of undefined"
This comes from the Node SDK and it is clear there is a parameter check missing.

 

Thanks to the debug I understood where the problem is.
Inside transaction.js file contained in node_modules/fabric-network/lib at row 235 we find:

 

const errorMessages = errorResponses.map((response) => util.format('peer=%s, status=%s, message=%s',const errorMessages = errorResponses.map((response) => util.format('peer=%s, status=%s, message=%s', response.peer.name, response.status, response.message));
So, the construction of the errorMessages suppose a peer is always online and makes no parameter check on the response. If the peer goes down, the response.peer parameter is undefined and what you get from response.peer.name is undefined too.
And this explains the error I get.

 

 

For a double check, I modified that line of code with:

 

const errorMessages = errorResponses.map((response) => util.format('peer=%s, status=%s, message=%s',const errorMessages = errorResponses.map((response) => util.format('peer=%s, status=%s, message=%s', response.peer? response.peer.name : 'offline', response.status? response.status : 'no status', response.message));
In this way, you check if the variables are undefined or not before sending a response, and this correctly work for my SDK that now returns:

"message": "No valid responses from any peers. 1 peer error responses:\n peer=offline, status=no status, message=Failed to connect before the deadline URL:grpcs://localhost:8051"
Please fix this bug in order to avoid SDK crash during this situation.

 

 

zhaochy1990 committed:
[FABN-1056] update return type of sign() to Buffer

current EC CryptoSuite sign() returns DER encoded signature
in array while the type definition for this is Buffer and the
jsdoc for api.CryptoSuite.sign() is byte[].

This CR updates the return type of sign() to Buffer.

Change-Id: Iea680c632b2465f552f3c7ae2181769bc6ce0572
Signed-off-by: zhaochy <zhaochy_2015@hotmail.com>
(cherry picked from commit cac1354f331a60d07dd449acf16ebff9d5fb7907)

bestbeforetoday committed:
FABN-1060: Handle error responses in comparison

Proposal responses returned from Channel.sendTransactionProposal()
can be any of: ProposalResponse, Error, or Error augmented with
response element of a ProposalResponse. Changing
Channel.compareProposalResponseResults() so that it can handle any
responses returned by Channel.sendTransactionProposal() without
error. Where any of the responses are not good (i.e. instances of
Error), compareProposalResponseResults() will return false, since
they don't contain valid write sets.

Similar change to Channel.verifyProposalResponse() to return false
for error responses.

If client code wants to compare only the successful proposal
responses, they can do so with:

const goodResponses = responses.filter((response) => !(response instanceof Error));
channel.compareProposalResponseResults(goodResponses);

Change-Id: Id1af8b3c5a1cb51eb2f4e9b50897671634848a8a
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
(cherry picked from commit ab1e3712ad0bfa54b66dc7a623151407c1af7778)

harrisob committed:
FABN-1067 NodeSDK prepare for 1.4.0

Prepare the branch release-1.4 for the release 1.4.0.

Change-Id: I2b67085fd22e3231adbd38aee0b515ccf65c5aa3
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FABN-1058 NodeSDK handler close peer

Have the handler close the peer connection
on error.

Change-Id: Iac956cb7e27fb792540976ce68057f1f5ba49937
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
(cherry picked from commit 89e8c4b14821ee9b28ecedb576521da2e86d9281)

harrisob committed:
FABN-1058 NodeSDK handler close peer

Have the handler close the peer connection
on error.

Change-Id: Ic43a2e16ad3e4ad792fb61b954fa121c39a445b4
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

rameshthoomu committed:
FABN-1079 Publish release-1.4 API docs

This updates the Publish API docs script to publish release-1.4
API docs.

Change-Id: I3a50732772fe4655ddb681fbcfa10b86d1dc8a61
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

harrisob committed:
FABN-1070 NodeSDK Use Node 8.14.0

Update build scripts to pull in node.js 8.14.0

Change-Id: I117f5f81b3205c7dda69bea1aa179261fd3ef872
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

rameshthoomu committed:
FABN-1081 Add new environment variable for logging spec

A new environment variable (FABRIC_LOGGING_SPEC) has been
added, and support for the CORE_LOGGING_LEVEL and log module
overrides has been removed.

FABN-1081 #done

Change-Id: I3e2c71b6eb7476a79eb413f4a4e298ce79c80419
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

andrew-coleman committed:
Merge "FABN-1070 NodeSDK Use Node 8.14.0" into release-1.4

andrew-coleman committed:
FABN-1084 X509 library locked to 0.3.3

Lock x509 npm module to v0.3.3

Change-Id: I0978da1c8d2b13986083b324497dc0093d29af90
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

bestbeforetoday committed:
FABN-1085: Minor doc corrections

Fixes to some types and bad links in the fabric-network JSDoc.

Change-Id: If60631933c365bc6ab1e8a8cfce5f490415393a2
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
(cherry picked from commit 5f7256f92c618ce95d76a5798ba98a16d1e509b7)

harrisob committed:
FABN-1058 NodeSDK add discover doc

Add some information on how users may call the channel
iniitialize again and again to provide a new target peer.

Change-Id: Ia92cdeba97b39f46b3ddb4d4869b55f777383b7e
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
FAB-13573 Prepare v1.4.0 release

Update version and tag properties in package.json for each package.
Update CHANGELOG.md

Change-Id: Ide5902bbd6cea528a67fd56a0a14d767145ff188
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

rameshthoomu committed:
FABN-1093 Update Jenkinsfile.z

This patch updates Jenkinsfile.z to remove pulling docker images
from nexus repository.

Change-Id: Ic497b92a9553e7853b12525ca4385c4cd7a28657
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

harrisob committed:
FABN-1083 NodeSDK allow commit/endorse handler

Document how to use the commit and endorse handlers.
The sendTransaction method will be enhanced to call
the commit handler when defined and initialized.
The sendTransactionProposal method will be enhanced to
call the endorsement handler when defined and initialized.

Change-Id: I579ecfe446886f92380e5ec125554d7147d4da43
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FAB-13574 NodeSDK prepare 1.4.1

Setup the branch to produce snapshots for the next
version of the npm packages.

Change-Id: Iefce93ac597ac0da235021002e16a6365b05d4ef
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
Merge "FAB-13574 NodeSDK prepare 1.4.1" into release-1.4

bestbeforetoday committed:
FABN-1156: Backport pluggable query handler

Change-Id: I7ea219d75fd2dacbd817f5c637ab11df6dfcfe6c
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

FABN-1069: TimeoutError on commit event timeout

Allows calling code to distinguish between transaction invocation
failure and a timeout waiting for a commit event following a
successful transaction invocation. Timeout errors can be identified
either by checking the error name or type:

- error.name === 'TimeoutError'
- error instanceof TimeoutError

Change-Id: I9a58bde13fbe2e2271680d9f5441b1b1de1712a8
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

FABN-1054: Private pool in EventHubFactory

Using channel.getChannelEventHub() to obtain event hubs in
EventHubFactory meant that users might inadvertently use the same
event hubs in their own code. This could potentially cause problems
with use of the fabric-network API if users performed a disruptive
action, such as using the event hub to replay previous block events.

Instead, maintain a private pool of event hubs in EventHubFactory,
with event hubs created using channel.newChannelEventHub().

Change-Id: If8ff4254500f22ea0ef4d4ff8732a440476aa0b1
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
FABN-1097 NodeSDK use class name

Avoid the use of instanceof as user may be using their
own location for the classes.

Change-Id: I42ceaea5238076850bbd3827259dc7eface7e6d5
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FABN-1098 NodeSDK allow multiple targets

Allow multiple targets on the channel initialize()

Change-Id: Ifb06d301c9ff516758df309613c09e5c654eff80
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
Merge "FABN-1083 NodeSDK allow commit/endorse handler" into release-1.4

mbwhite committed:
[FAB-13075] Move to updated x509

Change-Id: I8e8500f4c9f5b691a99932397a69dee920b72a86
Signed-off-by: Matthew B. White <whitemat@uk.ibm.com>

andrew-coleman committed:
Merge "[FAB-13075] Move to updated x509" into release-1.4

harrisob committed:
FABN-1096 NodeSDK add member_only_read

SideDB needs to support member_only_read attribute
on collection definitions

Change-Id: Ia79792359b17e13c2c736579ecb262498496e479
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

mbwhite committed:
FABN-1031 - additional grpc library for updating

An additional grpc library need to be updated

Change-Id: Ic52b6ad3271d5022e3cccfec9d85e7bb9b884120
Signed-off-by: Matthew B. White <whitemat@uk.ibm.com>

andrew-coleman committed:
Merge "FABN-1031 - additional grpc library for updating" into release-1.4

mbwhite committed:
[FABN-1031] Make grpc levels consistent

Change-Id: Ic7c2628ae5622448f00e73e60161a5f9222822fe
Signed-off-by: Matthew B. White <whitemat@uk.ibm.com>

rameshthoomu committed:
FABN-1110 Update CI script to remove nvm installation

This patch updates the CI script to remove the nvm installation
step.

Change-Id: Idd4d299336ceb60b2293b0ad46eccfd2f978f896
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

bestbeforetoday committed:
FABN-1124: Use MSP ID from client user context

If Client has a user context set, use the MSP ID of that user as
the default MSP ID for the client. If no user context is set,
use the MSP ID of the organization specified in the client section
of the common connection profile used to configure the client, as
before.

This allows the client section of the connection profile to be
omitted for discovery, and functions like Client.getPeersForOrg()
to return the correct results in this case.

Change-Id: I7e0f1bd79c904a0b665a3b411952beb65884e38e
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

GuillaumeCisco committed:
[FABN-1123] Fix identity update maxEnrollments

Move maxEnrollments key to max_enrollments key.

Change-Id: I3c233d55b919e82a2227f82f5dd77f74611f9082
Signed-off-by: GuillaumeCisco <guillaumecisco@gmail.com>
(cherry picked from commit 0a85a542a13328f42497aa9c0da4f5c2329589b9)

harrisob committed:
FAB-1117 NodeSDK event hub connect

Allow the connect to have a start block and
also be able to take a new peer target.
Add the reconnect to allow the application
to restart the connectioin and request
the start block to be the last block this
event hub has seen.
Restructure the testing to isolate all parts
to avoid conflicts between the different
tools.

Change-Id: I53bac25dd5c5e295b7cb94898a9a7f504cfd6fd2
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

liam-grace committed:
[FABN-1127] Corrected typescript definitions

- Updated ChannelEventHub.connect() definition
- Exported ChaincodeChannelEventHandle
- Upgrade typescript dependency

Upgrade the version of the typescript devDependency since the
previous version seemed to contain a duplicate definition for
the standard JavaScript SharedArrayBuffer type and was failing
to compile typescript. (Mark S. Lewis <mark_lewis@uk.ibm.com>)

Change-Id: I10d27d87bf83b1a94e7c5327e27efc108e0a3a4d
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

Commit:
[FABN-1130] Stop using "init" as default function name

The SDK currently forces you to supply a function name
to the instantiate command; if you don't, it defaults
to "init". There is no need to specify a function name
during instantiate, and the new programming model does
not need you to call *any* function at instantiate time.

This is the v1.4 backport; in order to maintain compat
with existing SDK applications, this change will keep
the current behaviour if you don't pass "fcn", but will
change the behaviour if you pass "fcn" with a null or
empty string value (highly unlikely anyone is doing that).

Change-Id: Ia526d855f66fc428bca43786bc4555e9da074a41
Signed-off-by: Simon Stone <sstone1@uk.ibm.com>

andrew-coleman committed:
FABN-1128 Discovery options not merged correctly

Fix to correctly merge the options structure passed into
the Gateway connect() method

Change-Id: Ifbdf604246c9cfcbfc893fab8f5284ef9b4faa19
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

harrisob committed:
Merge "FABN-1128 Discovery options not merged correctly" into release-1.4

bestbeforetoday committed:
FABN-1143: Remove spurious error log

Calling Client.getChannel() with the option to prevent errors
still resulted in error log being produced. Changed to if
errors are suppressed a debug log is produced instead of an
error.

Change-Id: Ib074a05f46cf15932551239008e103aad2c42f18
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

Commit:
[FABN-1155] Include missed connection setting

Added settings from fabric-client/default.json
Updated unit-tests

Change-Id: Ibc1cdd7a571a7555c498489ca6024d144ded75aa
Signed-off-by: Leo Le <leo2le@protonmail.com>

bestbeforetoday committed:
FABN-1118: Unit test fixes for Node 10

Two unit tests were checking for specific error message text
when passing a non-string argument to the Node url module's
legacy parse function. This error message looks to have
changed. Fixing the tests to check for an expected error type
rather than specific text produced from code we don't own.

Change-Id: Id890f3084ac1e26ca832150ead8e5001614a6f32
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

andrew-coleman committed:
Merge "FABN-1156: Backport pluggable query handler" into release-1.4

harrisob committed:
FABN-682 NodeSDK  client specify escc/vscc names

After Fabric v1.2 released, new features for
pluggable endorsement and validation was introduced.
Thus node sdk users may want to use their own
escc/vscc other than the buildin.
Set the default values so this change will also cover
FABN-1126 where these names must be specified to work
with fabric 1.1

Change-Id: Iba72692ef1b7506da62190a392b296743ad2744c
Signed-off-by: zhaochy <zhaochy_2015@hotmail.com>
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

Commit:
[FABN-1159] Allow PEM paths relative to CCP

Allow a connection profile to specify paths to PEM files (for peer,
orderer, and certificate authority  TLS CA certs) that are relative
to the connection profile file. Currently they must be relative to
the current working directory, which is kind of useless if you want
to zip up a connection profile and the PEM files and ship them around.

Change-Id: Icd23c90553e5e4c81f6fe9245b6740b83749ec3e
Signed-off-by: Simon Stone <sstone1@uk.ibm.com>

Commit:
[FABN-1160] Fix TLS logic for discovered peers

The SDK assumes by default that peers and orderers
returned by service discovery will be running with
TLS (grpcs://). This is a bad assumption when we
are connecting to a gateway peer that is not running
with TLS (grpc://).

Change the logic so that if we connect to a peer
to perform discovery over TLS, then we assume that
the discovered peers and orderers also use TLS.
Likewise if we connect to a peer to perform discovery
without TLS, then assume that the discovered peers
and orderers also do not use TLS.

It is highly unlikely (impossible even?) that you
can have a mixed Fabric network with TLS and non-TLS
peers.

Change-Id: I9b8e0056cd6c6c743d6d7feab561cdaa8f7d3f05
Signed-off-by: Simon Stone <sstone1@uk.ibm.com>

nklincoln committed:
[FABN-1169] Backport certifcate fix from master

- Backport fix from master
- Enable mutual TLS (client authetication)

Change-Id: Id2fab00875f643e7734458158f3a5dddcf6f5360
Signed-off-by: nkl199@yahoo.co.uk <nkl199@yahoo.co.uk>

sambhavdutt committed:
FABN-1177 Add gulp end-to-end test for fabric

Add end-to-end gulp test, to verify the fabric
e2e-daily and e2e-merge.

Change-Id: Ifbb26b61fe78d68e70607526e1de39aa76512ee1
Signed-off-by: Sambhav Nidamarty <sambhavdutt@gmail.com>

liam-grace committed:
[FABN-1100] Event handling abstractions

- Handle commit, contract and block events
- Checkpoint each event listener to allow easy event catch up
- New event hub manager to centralise all event hubs used by fabric-network
- Automatic reconnect to an event hub when a registered listener loses connection

Change-Id: I93f3df089e5393fbaf90592528463f1ee5d79b1d
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

rameshthoomu committed:
FABCI-304 Update CI Pipeline script

This patch updates CI pipeline scripts which utilizes global shared
library reusable functions. It also creates a ci.properties file with
all the parameters required to test the fabric-sdk-node patch.

Change-Id: I0d93498deacc8a784272f5c47b1a87c831118db6
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

mastersingh24 committed:
Properly handle empty responses

Obtain the HTTP status code from the
response object rather than the response
body.

FABN-1187 #done

Change-Id: Idf0401f760f8d2dcaaf51ed4bee1e00c9b04c3ff
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

sambhavdutt committed:
FABN-1188 Update thirdparty version to 0.4.15

Update the testFabricThirdParty version to 0.4.15

Change-Id: I74284e9ba4402e2fb3ee924b0112d3ce9ed67984
Signed-off-by: Sambhav Nidamarty <sambhavdutt@gmail.com>

harrisob committed:
Merge "FABCI-304 Update CI Pipeline script" into release-1.4

rameshthoomu committed:
FABN-1193 Update CI Scripts

This patch fix the errors observed in fabric-sdk-node
merge jobs. Also, update the documentation.

Change-Id: I1084a12a2c585766006d4810c60f34d9497c5e8f
Signed-off-by: rameshthoomu <rameshbabu.thoomu@gmail.com>

andrew-coleman committed:
Merge "[FABN-1100] Event handling abstractions" into release-1.4

liam-grace committed:
[FABN-1100] Ammended typescript definitions

Change-Id: I46bead0a9f3da9457c3bad3d6c2adddda3768ae8
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

Commit:
[FABN-1194] Use connection options for discovery peer

Make Client.buildConnectionOptions public and use Clients's\default connection options
for creating discovery peer

Change-Id: I497d746cca72b6e7b7064ecc30e8fabf46cfdfae
Signed-off-by: LeoLe <leo2le@protonmail.com>
(cherry picked from commit 53359f982ce1209e0c1ef7c7a882ac45ab611ebf)

liam-grace committed:
[FABN-1100] Checkpoint after callback is complete

Change-Id: I0251af6ca3a8559baac09c458a28173c91ed752a
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

andrew-coleman committed:
Merge "[FABN-1100] Checkpoint after callback is complete" into release-1.4

liam-grace committed:
[FABN-1100] Tidy start block and event hub management

Change-Id: I44b3683283fa2d13404ee15d109af7fc6dd2159a
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

andrew-coleman committed:
Merge "[FABN-1100] Tidy start block and event hub management" into release-1.4

andrew-coleman committed:
FAB-14976 Release fabric-sdk-node v1.4.1

Created release note
Updated CHANGELOG
changed verision and tag properties in the package.json files

Change-Id: I8cbbb75aabf8ad949679720ba14492a27a2c1433
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

harrisob committed:
FABN-1201 NodeSDK fix SO_TIMEOUT error

Test was getting a different error message than
expected.

Change-Id: Iefd9f97638db60cb9ade9c48f5afb7c33e4f52c6
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
(cherry picked from commit d6b9cb1f94828d907de8b66c6c2ec187dde96509)

harrisob committed:
FAB-14977 NodeSDK prepare for next release

Update all version to prepare for the next
release

Change-Id: I0c158227442d6c3cd75befef8b3772ed9d2bd2b4
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
Merge "FAB-14977 NodeSDK prepare for next release" into release-1.4

harrisob committed:
FABN-1207 NodeSDK chaincode event defaults

Chaincode event default for unregister  would cause the
event listener to be disconnected after one event was
received when setting endBlock. Now for all event listeners
the default to unregister will be true when the endBlock is
received. Default for disconnect will be disconnect the
channelEventHub when the endBlock is seen.

Change-Id: If7ef15674411d52c15eaa7554dfd8baca945d1e3
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

liam-grace committed:
[#FABN-1206] Fixed inability to receive filtered events

Change-Id: Ic1588250452a8502fedb53b53a95328dcf3b194f
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

harrisob committed:
FABN-1210 NodeSDK Chaincode events as array

Allow all the chaincode events found in a block to be
sent  at one time rather than sent  one at a time.
This will be controlled by a new setting on the chaincode
listener registration, "as_array", the default will
be to send as before, one at a time.

Change-Id: I1dd244eea8743b0394f0feee77068cc1582763bd
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

zhaochy1990 committed:
[FABN-1216] package.json add node 10

update the package.json engines to allow node 10.

Change-Id: Ia4f2cf3951a68618cb07d076cfcc90044dc9ca78
Signed-off-by: zhaochy <zhaochy_2015@hotmail.com>

liam-grace committed:
[FABN-1100] Fix for checkpoint factory

Change-Id: Ic1e88d3daa95fd486be7076c2f0e85d36507c48d
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

harrisob committed:
Merge "FABN-1207 NodeSDK chaincode event defaults" into release-1.4

andrew-coleman committed:
Merge "[FABN-1100] Fix for checkpoint factory" into release-1.4

Commit:
[FABN-1225] Convert CDS path seps from Win to UNIX

Convert the path separators in the path we store in
the chaincode deployment spec from Windows ('\') to
UNIX ('/'), as the Go chaincode build process uses
this path and that runs on UNIX.

Note that the paths in the tarball are already being
automatically converted by existing code.

Quite difficult to unit test this without any Windows
systems to run unit tests on!

Change-Id: I7f149482512541c850bbb8b6259bc4e4e5df9815
Signed-off-by: Simon Stone <sstone1@uk.ibm.com>

bestbeforetoday committed:
FABN-1226: Better handling of query error responses

Query handlers throw on peer error responses instead of treating
them as communication failures and querying additional peers.

Updated TypeScript definitions to include information required
for client code to make the same distinction, which was already
included in the actual JavaScript objects.

Change-Id: Ia0f2f2d763ac7b5f0fdd421838e2180a0bddf323
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
FABN-1231 NodeSDK add connect to event tutorial

Update the channel event hub tutorial to add info about
the flow of the different event types and how to use
the connect call.

Change-Id: Ia5f60d83cfaa85aaa41366b5e9aceffc37c5fd56
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

davidkhala committed:
[FABN-1232] missing gulp test

Reported from Hyperledger internship students,
gulp test is undefined at this moment
It will be better to revert it since it is widely used among documents.

Change-Id: I886ce3503e05deb8acaf8d3a70e7b5d40c57783b
Signed-off-by: davidLiu <david-khala@hotmail.com>

andrew-coleman committed:
Merge "FABN-1231 NodeSDK add connect to event tutorial" into release-1.4

liam-grace committed:
[FABN-1205] Chaincode event data fix

- Documentation and typescript definition changes
- as_array fabric-client.ChannelEventHub option used when listening for contract events

Change-Id: I74f78e9c12147046c108158b039edd0ea3cacbd3
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

liam-grace committed:
[FABN-1235] Block event checkpointer fix

- Fixed issue where blockNumber in _onEvent:blockeventlistener.js is always
  NaN when receiving unfiltered block events

Change-Id: I9088648c0b73cd7df61b4f7d49efe9c76e87b330
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

bestbeforetoday committed:
FABN-1239: Better error on submitTransaction failure

Improve the error message produced when
Transaction.submitTransaction fails because of bad proposal
responses. Include the peer name and response status in the
error message.

Change-Id: I683a7686f39668ddfdbe55955431c1a5203f4977
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

andrew-coleman committed:
Merge "FABN-1210 NodeSDK Chaincode events as array" into release-1.4

harrisob committed:
FABN-401 NodeSDK Add connect info to tutorial

Update the connect calls to not show the boolean.

Change-Id: Ia2d5d7ba1974fc1c9a7126ec73eb1b8019e9d22f
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FABN-1133 NodeSDK switch to caname

Use  caname on all fabric-ca outbound calls to line upe
with Fabric-CA use of the name.

Change-Id: If008ecb0a25ebbbfe00ca11d754e338d2fa58581
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
Merge "FABN-1133 NodeSDK switch to caname" into release-1.4

liam-grace committed:
[FABN-1203] fabric-network event listening doc changes

Change-Id: I2832c4e8b68f966967dd78540435c3cdac25e1a1
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

harrisob committed:
FABN-1256 NodeSDK reset as_local default

Set the default as_local to false for discover to
align with production systems.

Change-Id: I555c3148dfa29263d78d1de43495c7288865edb3
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FABN-1220 NodeSDK fabric-ca-client dependency

Have the fabric-ca-client as a dpendecy for fabric-client.

Change-Id: I599d2cca47fddb2e8f7a381d3d3289106046b39e
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

zhaochy1990 committed:
Merge "FABN-1220 NodeSDK fabric-ca-client dependency" into release-1.4

harrisob committed:
FABN-1220 NodeSDK fix fabric-ca-client version

Fix the publish script to modify the fabric-ca-client version
to the current version in the fabric-client package.

Change-Id: I2c67fc540ae533eb97c7b1ab6d486a7e1ef2771d
Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
Merge "[FABN-1203] fabric-network event listening doc changes" into release-1.4

andrew-coleman committed:
FABN-1259 fix timeout when debugging chaincode

If the commitTimeout property is larger than the ‘request-timeout’
then invoke ‘sendTransactionProposal’ with this larger timeout

Change-Id: Ic1596cfb93368c430d51afafce01f5adc3601a8c
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

harrisob committed:
Merge "FABN-1259 fix timeout when debugging chaincode" into release-1.4

liam-grace committed:
[FABN-1251] Added start/end block functionality

- Ability to specify startBlock and endBlock when using fabric-network to handle events
- Added configurable event hub timeout when creating listeners
- Added configurable wait time between trying new event hubs
- Updated documentation
- Updated typescript definitions

Change-Id: Ifca2d775e6b5f9d583ddb7bd5b81d9b8f0018bcb
Signed-off-by: Liam Grace <liamgrace.896@gmail.com>

andrew-coleman committed:
FABN-1275 Publish v1.4.2 maintenance release

Change-Id: I3f728698213983eb9219d32864e84863f3652396
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

harrisob committed:
FABN-1220 NodeSDK add fabric-ca-client

Add fabric-ca-client dependency to fabric-client
when a release with the proper version.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: If7b78d170fd3f2a3015b81e278b61df2d521bf07

Commit:
[FABN-1282] Fix integration test by README and code.

This pull request fixes three parts of integration test of
release-1.4 branch:

1. Remove --with-default-names for macOS in README.
2. Fix on logger.js and contracteventlistener.js warnings caused by no-console.
3. Fix on AssertionError in Peer.js.

**Note: this is a task about fixing integration test from
this year's Hyperledger Mentorship Program, fabric node
sdk security extension project.

I have run integration test by `gulp test` on my mac.

For references and change log please see
https://jira.hyperledger.org/browse/FABN-1282.

Signed-off-by: “5sWind” <fiveswind@gmail.com>
Change-Id: Id0cae42101724951ece0c05d6809555fa031bf68

andrew-coleman committed:
FABN-1284 1.4.3-snapshot update

Change version to 1.4.3-snapshot in package.json files

Change-Id: I76f6a1798c9d254a74e6239e43da486b5133cb53
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

andrew-coleman committed:
Merge "[FABN-1251] Added start/end block functionality" into release-1.4

harrisob committed:
FABN-1288 NodeSDK remove clientKey from log

All GRPC options were placed into log for debug,
the clientKey is not required and should not
be displayed.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: Iec6cc5c2fa3d3e3390a3556066e43b0567d6421d

mastersingh24 committed:
Merge "FABN-1288 NodeSDK remove clientKey from log" into release-1.4

andrew-coleman committed:
FABN-1303 Update to version 1.21.1 of grpc

Change-Id: I8d6390db201ac65bf8b887774f9761ef10624930
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

andrew-coleman committed:
FABN-1305 Publish v1.4.4 maintenance release

Change-Id: I2d404fdc3621a22753e987a98676c9e4e0e63759
Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

bestbeforetoday committed:
FABN-1306: Set versions to 1.4.5-snapshot

Change-Id: I5a6a263428da5d6132f8b04fc5155a53c328849b
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

bestbeforetoday committed:
FABN-1322: JSDoc for DefaultEventHandlerStrategies

Correctly describe the different default commit handler behaviours.

Change-Id: Ifb1277c0a155bce01cb67392cea20b9e00904f1e
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
FABN-1325 NodeSDK update readme for Node.js vers

Update readme to show versions of Node.js supported

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: Ia6efdac8381bd4649060b461ee041eab55433e45

bestbeforetoday committed:
FABN-1326: Use only eventing peers for commits

Only peers in the eventSource role should be used to listen for
transaction commit events.

Change-Id: Ibdc7396f66123154c4c5a4db6fe774b05d6a02f0
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

mbwhite committed:
[FAB-16105] Update build.gradle for Java Contracts

Change-Id: Ieba68c9a3a7ca38b1344f1835440c1e446efb329
Signed-off-by: Matthew B. White <whitemat@uk.ibm.com>

davidkhala committed:
[FABN-1334] clean duplicated code and jsDoc

As titled

Signed-off-by: david <david-khala@hotmail.com>
Change-Id: Ic942a8b3b33c4ee29af92f5c4361802b8cdd933d

bestbeforetoday committed:
FABN-1332: Revert commit handler implementation

Revert the implementation of the default commit handlers to
equivalent to the 1.4.0 implementation to restore performance.
This means using event hubs directly to listen for commit events
rather than using Transaction.addCommitListener().

Also update the sample custom sample transaction event handler
to use cached event hubs rather then new event hub instances on
each transaction for better performance.

Change-Id: I2f5e799f8983ef688c0351d1153aeed2ab173de6
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
Merge "[FABN-1334] clean duplicated code and jsDoc" into release-1.4

harrisob committed:
FABN-1233 NodeSDK segment fault

Segment faults with Node.js are typically with the
native code of a package. GRPC has native code. We
will upgrade the grpc package.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: Iece067301bc4523baea4074ba1fc866f6f4293b7

heatherlp committed:
[FABN-1320] Add git_tag.sh script

-  for use in manual release process

Signed-off-by: heatherlp <heatherpollard0@gmail.com>
Change-Id: I44eca5b929b613bd618cdaaf5393581c854fb44a

Commit:
[FABN-1341] Support .fabricignore files in packaging

The packaging code in the SDK for Fabric v1.4 lacks
general purpose options for controlling which files
in a directory get packaged into a CDS file. Node.js
has the option to handle a .npmignore file, but the
other platforms do not have the same option. This is
a particular problem with Java, because the build
directories containing uber JARs end up in the CDS
package, then get dropped the other end.

This CR adds support for an optional .fabricignore
file that can be placed in the contract directory.
This file can be used to ignore any files desired
by the user. In the case of Go and Node, which
ignore non-Go files and node_modules respectively,
this behaviour is disabled in the presence of a
.fabricignore file, giving the user full control
over what gets packaged.

Signed-off-by: Simon Stone <sstone1@uk.ibm.com>
Change-Id: I362d786733e2d8f8b12406c20ef02b66b1c6787d

Commit:
[FABN-1342] Improve SDK packaging performance

Use custom writable stream implementation instead
of inefficient stream-buffers package when building
smart contract packages.

Signed-off-by: Simon Stone <sstone1@uk.ibm.com>
Change-Id: I51de2aa6cc62933340dd31e6b3af36f376c6f646

andrew-coleman committed:
Merge "FABN-1325 NodeSDK update readme for Node.js vers" into release-1.4

bestbeforetoday committed:
FABN-1344: Allow endorsing peers to be specified

Change-Id: I552000af321b40fd35e14fef787b3c41e0de0336
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

Commit:
[FABCI-403] Disable AnsiColor Wrapper

Signed-off-by: Brett Logan <Brett.T.Logan@ibm.com>
Change-Id: Ia365f7c485972abc66c5b6b20dd1e3ec02adc04f

harrisob committed:
FAB-10717 NodeSDK protect endorsement plan

Make a deep copy of the endorsement plan to avoid
changes and additions added during runtime from
bleeding into other users.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: I2e66b8b6bedf1f97b70820985d418cbfbae6801f

harrisob committed:
FAB-10717 NodeSDK protect endorsement plan

Make a deep copy of the endorsement plan to avoid
changes and additions added during runtime from
bleeding into other users.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: Ic1367153380f50cee3e4e80a86b8df631e2e042c

harrisob committed:
FABN-1348 NodeSDK Define discovery

Define the discovery settings for Network testing.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: I22fce39edafb9f4f898e988a3a1db12fdfb0f9d1

Vijaypunugubati committed:
FABN-1358 Update FabricVer to pull latest images

Updated the Fbaricversion to pull latest 1.4.3
images from docker.

Signed-off-by: vijaypunugubati <vijaypunugubati@gmail.com>
Change-Id: If661cdc6a0de05b5a535b44ff4ef4b8bc13ac1a5

Commit:
[FABN-1359] Move HSM docs into separated page

Will move and prettify HSM detailed contents in README.md into
fabric-sdk-node.github.io tutorials, since the special tests
for HSM is not necessary thing for every builder/developer.
Let's move it out of README.

Signed-off-by: “5sWind” <fiveswind@gmail.com>
Change-Id: Ia97289b27c5f6efae4ca2e03ea7a1cf6ee56010a

harrisob committed:
Merge "[FABN-1359] Move HSM docs into separated page" into release-1.4

andrew-coleman committed:
Merge "FABN-1348 NodeSDK Define discovery" into release-1.4

harrisob committed:
FABN-1384 NodeSDK update grpc level

Update grpc to 1.23.3

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>
Change-Id: Ica524ed280026b023a737a22aee8437b1f0b8a7c

ryjones committed:
[IN-68] Add default GitHub SECURITY policy

this file must exist on the default branch

Signed-off-by: Ry Jones <ry@linux.com>
Change-Id: I2ea1dde43e2b084cf615d8a6597b631d854119b0

lesleyannjordan committed:
[FABN-1335]-Reference event handling

Signed-off-by: lesleyannj <lesleyannj@hotmail.com>
Change-Id: I843be29b366c10e9943be310b53f83fa54481331

denyeart committed:
fabric-sdk-node update to baseimage 0.4.16

Change-Id: I3631b362d2a4a2701cdb8bb3825eb180d582c54c
Signed-off-by: David Enyeart <enyeart@us.ibm.com>

davek-at-ibm committed:
FABN-1333

Improve error messages if channel fails to initialise
due to no peers to initialise from

Change-Id: I64e8195ff666bd058e31cd3a104d376aa0537298

Signed-off-by: Dave K <d_kelsey@uk.ibm.com>
Change-Id: Icefcb2b64e8e329fe3fcec9297f7a57c499d3704

davek-at-ibm committed:
FABN-626 make pkcs11js optional

So long as no further native modules are
introduced, the node-sdk should install onto
windows systems without having to install
windows compilers

Signed-off-by: Dave K <d_kelsey@uk.ibm.com>
Change-Id: I0fa4ab5e3380bb57e9394ea60903b0a35c89c6c4

heatherlp committed:
[FAB-16680] Fix jitpack link

- for verify build failures

Signed-off-by: heatherlp <heatherpollard0@gmail.com>
Change-Id: I7a3425e339c66a85f904247431397f040ce417bd

andrew-coleman committed:
Merge "FABN-1333" into release-1.4

Commit:
[FABN-1400] Clean up cryptoSuite key signing.

No longer need sign opts in newer versions.

1. identity.js
2. pkcs11.js

Signed-off-by: Hengming Zhang <fiveswind@gmail.com>
Change-Id: Ieff8c8e77c2e7e12152b1a2d838edf044a4b2bdb

andrew-coleman committed:
Merge "fabric-sdk-node update to baseimage 0.4.16" into release-1.4

Commit:
[FABN-1402] Use "for of" instead of "for in"

Avoid Config.reorderFileStores failure when
this._fileStores array has a property called
"choose" (assume from dodgy polyfill) by using
"for of" instead of "for in". As a bonus, this
makes the code slightly simpler.

Signed-off-by: Simon Stone <sstone1@uk.ibm.com>
Change-Id: I0c4e16045aaa90da28ac014d6e84efd4534a3f4d

harrisob committed:
Merge "[FABN-1400] Clean up cryptoSuite key signing." into release-1.4

andrew-coleman committed:
Merge "[FABN-1402] Use "for of" instead of "for in"" into release-1.4

bestbeforetoday committed:
FABN-1386: Azure pipelines build

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Change-Id: If19c3a190b145c0f12236f47ae6725053c6fe3a1

bestbeforetoday committed:
FABN-1386: Publish API documentation in Azure

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Change-Id: I54802709f3772e7b998f7bd2bba5e593dcd0ef94

bestbeforetoday committed:
FABN-1386: Azure Pipelines publish of npm packages

Publishing is currently commented out in the Azure Pipelines
script until the required publishing tokens are available.

Also slight tidy-up of API documentation publishing script.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Change-Id: I281f8968a5606d10bfc8966ad8dc400357a7c327

bestbeforetoday committed:
FABN-1386: Don't pull Docker images

Release 1.4 uses release versions of Fabric Docker images. It does
not need to pull development snapshots.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

Commit:
FABN-1419: Fix HSMWalletMixin typescript definition

Signed-off-by: Liam Grace <liam.grace@ibm.com>

heatherlp committed:
[FAB-1418] Fix builds and add package-locks
- rearrange build.grade setup to avoid jitpack woes

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

harrisob committed:
Merge pull request #33 from heatherlp/fixCATests

[FABN-1418] Fix builds and add package-locks

bestbeforetoday committed:
FABN-1386: Publish docs to gh-pages branch

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

andrew-coleman committed:
Merge pull request #30 from liam-grace/issue-1419

FABN-1419: Fix HSMWalletMixin typescript definition

andrew-coleman committed:
Merge pull request #35 from bestbeforetoday/FABN-1386-1.4

FABN-1386: Publish docs to gh-pages branch

Commit:
[FABN-1407] Add PDC and EP types (#28)

Add typings for a private data collection configuration
and endorsmeent policy to ChaincodeInstantiateUpgradeRequest.

Signed-off-by: Simon Stone <sstone1@uk.ibm.com>
Change-Id: I32f1789dbb96fecc88d03fcd841356d21fcb89e9

bestbeforetoday committed:
FABN-1386: Reference credentials in pipeline script

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
Merge pull request #41 from bestbeforetoday/FABN-1386-1.4

FABN-1386: Reference credentials in pipeline script

bestbeforetoday committed:
FABN-1386: Publish correct doc release directory (#42)

Invoke documentation generation with the correct branch name so
that built documentation is published under the correct
sub-directory for the current branch. Previously, all branches
published their documentation as master.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

bestbeforetoday committed:
FABN-1386: Doc publishing fixes (#43)

- Git and find usage tweaked for cross-platform compatibility
- Ensure publishing script fails fast on any errors
- Add some additional information output

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

bestbeforetoday committed:
FABN-1386: Publish npm packages

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
Merge pull request #44 from bestbeforetoday/FABN-1386-1.4

FABN-1386: Publish npm packages

bestbeforetoday committed:
FABN-1386: Tidy up npm publish script (#45)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

heatherlp committed:
(Release-1.4) Remove gerrit/jenkins references (#49)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

liam-grace committed:
[FABN-1307] Pass options.unregister through to clientOptions.unregister (#54)

Signed-off-by: Liam Grace <liam.grace@ibm.com>

harrisob committed:
FABN-1403 NodeSDK modify multicert pems (#56)

Modify pem encoded certificates if including mulitple
certificates and separated by new lines

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

bestbeforetoday committed:
FABN-1386: Import npm credentials in build (#63)

npm credentials must be imported to allow publishing to npm as part
of the merge builds.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

liam-grace committed:
[FABN-1308] Enable checkpointing when start or  end block is given (#52)

Signed-off-by: Liam Grace <liam.grace@ibm.com>

heatherlp committed:
(Release-1.4) Add Mark Lewis as a maintainer (#77)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

harrisob committed:
FABN-1403 NodeSDK add new line to certs (#89)

In the case where a certificate chain is not separated by
a new line, one will be added.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

harrisob committed:
FABN-1450 NodeSDK publish 1.4.5 (#90)

Update package.json to publish 1.4.5

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

Commit:
[FABCI-482] Update Nexus URLs to Artifactory (#82)

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

bestbeforetoday committed:
FABN-1446: Specific compatibility matrix for 1.4 (#92)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

harrisob committed:
FABN-1460 NodeSDK set release 1.4 snapshot (#100)

Update the package.jsons to allow for fixes and development
of the 1.4.6 of release-1.4 branch.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

heatherlp committed:
[FABN-1039] Cherry pick of changes delivered under "Create eslint task" (#101)

From master, commit hash is 67365b5ea7a5319f26c59e500b2420b61af95dc4

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp commented:
Hey @rbntt I think somethings gone awry with your commits. It looks like you're trying to merge release-1.4 into master, which we definitely don't want to do :) 
If you get the branches sorted out, I can review
Thanks

bestbeforetoday commented:
The peer details should always exist so it looks like there may be a code path that fails to set them. A better fix might be to ensure they are set.

harrisob committed:
FABN-1463 NodeSDK make waitForReady optional (#103)

The gRPC setup 'waitForReady' on services is an optional setup
step on a service connection that could be bypassed when not
required by the application.

Allow for a connection option to skip and move straight to
sending the outbound request.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

andrew-coleman committed:
FABN-1451 add peer responses to error (#108)

In submitTransaction(), if there are no valid responses from peers, an error is thrown back to the client.
This PR adds the array of responses to the error object.

Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

heatherlp committed:
Fix some broken docs links (#112)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

andrew-coleman committed:
add jsdoc for peer responses in exception (#113)

Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

bestbeforetoday committed:
[FABN-1430] Fix type for IServiceResponse (#62) (#114)

The name of the members of IServiceResponse should start with small
letters as per lib/serverendpoint.go in fabric-ca.

Cherry-pick of 0c6f22965f672538257dd2508f1be57df6bdb6c6

Signed-off-by: Taku Shimosawa <taku.shimosawa@hal.hitachi.com>
Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

Co-authored-by: shimos <taku.shimosawa@hal.hitachi.com>

heatherlp committed:
[FABN-1039] Remove gulp from release1.4 (#107)

* [FABN-1039] Remove gulp from release-1.4

Initial commit after manually applying PR#61

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

* [FAB-1039] Additional work required for gulp removal

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

* FABN-1039: Fix master branch doc publishing (#74)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

* [FAB-1039] PR #107 Clean up api docs publishing scripts

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

Co-authored-by: Mark S. Lewis <Mark.S.Lewis@outlook.com>

heatherlp committed:
[FABN-1039] Port dependencies clean up to release-1.4 (#117)

* [FABN-1039] Port dependencies clean up to release-1.4

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

* [FAB-1039] Fixes for newer eslint rules

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1481] Remove node 8 support and add node 12 (#118)

- Run node 10 and 12 sequentially
- Fix some strings in the tests
- Update package package.jsons
- Fixed typos

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1487] Release 1.4.6 (#140)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1487] Revert release-1.4 branch to release snapshots (#141)

- Also added .npmignore file to fabric-network, which currently publishes test files

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1500] Fix fabric-ca-client builds (#142)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
Release v1.4.7 (#143)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
Revert to releasing snapshots after 1.4.7 (#146)

- Also upped the version of fabric we test with to 1.4.6, released yesterday

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1505] List winston as dependency for all packages (#153)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

bestbeforetoday committed:
FABN-1481: Add Node 12 to compatibility matrix (#155)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

davek-at-ibm committed:
FABN-1508 setEndorsingOrganizations support (#163)

Signed-off-by: Dave K <d_kelsey@uk.ibm.com>

add setEndorsingOrganizations support

heatherlp committed:
FABN-1510 Update grpc levels to 1.24.2 (#164)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

bestbeforetoday committed:
Trigger for PR builds (release-1.4) (#174)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

louis-murray committed:
FABN-1515: consume CKS as a synchronus function (#172)

* fix: consume CKS as a synchronus function
* use async/await

Signed-off-by: Louis Murray <louis.murray@ibm.com>

heatherlp committed:
FABN-1517 Publish 1.4.8 (#177)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
FABN-1517 Fix release-1.4 tags for 1.4.8 release (#179)

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
FABN-1517 Revert to 1.4.9-snapshots (#178)

- Also fix recurring test failure in fabric-client/ChannelEventHub

Signed-off-by: heatherlp <heatherpollard0@gmail.com>

heatherlp committed:
[FABN-1257] Fix EventHub error when reconnecting (#186)

Fix timeout error that occurs when we immediately
reconnect EventHub. The cause of this problem is
inappropreate handling of incomming reset message
for the disconnected stream.

Change-Id: Ifa9802b7c5e71124ea6ec3ed3c4afd73102bce26
Signed-off-by: Yohei Ueda <yohei@jp.ibm.com>
(cherry picked from commit 60d160dbdf098acd99ec3d3f636fd4987fe2ee60)

Co-authored-by: Yohei Ueda <yohei@jp.ibm.com>

harrisob committed:
FABN-1224 NodeSDK discovery interest 1.4 (#132)

Add discovery interest to fabric-network release 1.4.
Setup a new method on Contract to take the dicovery
hint and pass it to the fabric-client channel.sendTransactionProposal.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

heatherlp commented:
Hi @rbntt are you planning on fixing up this PR? All participants are getting notified every time a new commit is merged, due to this PR trying to merge release-1.4 and master. Can I recommend that we close this one and you re-open another just containing your commit? Happy to talk you through that if it helps
